opentelemetry-user-events-logs/src/logs/exporter.rs

```diff
@@ -13,15 +13,38 @@ use std::{cell::RefCell, str, time::SystemTime};
 
 thread_local! { static EBW: RefCell<EventBuilder> = RefCell::new(EventBuilder::new());}
 
-/// UserEventsExporter is a log exporter that exports logs in EventHeader format to user_events tracepoint.
-pub(crate) struct UserEventsExporter {
+/// Trait for event name callback
+pub trait EventNameCallback: Send + Sync {
+    /// Returns the event name for the given log record
+    fn get_name(&self, record: &opentelemetry_sdk::logs::SdkLogRecord) -> &'static str;
```

Comment:
Returning static str has implications. Why not `Cow<'static, str>` - sorry if it was already agreed on earlier. But good to mention here if someone want to track the history. 
nit - Also, how about returning Option, as the user provided callback can fail too. But this is not necessary, as callback should return default rather than failing. But just adding if any discussion required.